command/pki_healthcheck.go

```diff
@@ -0,0 +1,1106 @@
+	"/config/crl": {vault.CreateCapability, vault.UpdateCapability},
+	"/config/url": {vault.CreateCapability, vault.UpdateCapability},
+	// Rotating CRL isn't privileged.
+	"/intermediate/generate/internal": {vault.CreateCapability, vault.UpdateCapability},
+	"/intermediate/generate/external": {vault.CreateCapability, vault.UpdateCapability},
+	"/intermediate/set-signed":        {vault.CreateCapability, vault.UpdateCapability},
+	// Revoke is privileged since it is by serial number and not by proof
+	// of possession.
+	"/revoke":                 {vault.CreateCapability, vault.UpdateCapability},
+	"/roles/":                 {vault.CreateCapability, vault.UpdateCapability, vault.DeleteCapability},
+	"/root/generate/internal": {vault.CreateCapability, vault.UpdateCapability},
+	"/root/generate/external": {vault.CreateCapability, vault.UpdateCapability},
+	"/root/sign-intermediate": {vault.CreateCapability, vault.UpdateCapability},
+	"/root/sign-self-issued":  {vault.CreateCapability, vault.UpdateCapability},
+	"/sign-verbatim":          {vault.CreateCapability, vault.UpdateCapability},
+	"/tidy":                   {vault.CreateCapability, vault.UpdateCapability},
+}
+
+var (
+	_ cli.Command             = (*PKIHealthCheckCommand)(nil)
+	_ cli.CommandAutocomplete = (*PKIHealthCheckCommand)(nil)
+)
+
+func maxInt(a int, b int) int {
+	if a > b {
+		return a
+	}
+
+	return b
+}
+
+func sortedKeys(checks map[string]PKIHealthCheckHelper) []string {
+	var result []string
+	for name := range checks {
+		result = append(result, name)
+	}
+	sort.Strings(result)
+	return result
+}
+
+type PKIHealthCheckCommand struct {
+	*BaseCommand
+
+	crlConfig map[string]interface{}
+	roles     map[string]map[string]interface{}
+	issuer    *x509.Certificate
+
+	mountDefaultTTL        time.Duration
+	mountMaximumTTL        time.Duration
+	fetchedMountAuditKeys  bool
+	mountAuditRequestKeys  []string
+	mountAuditResponseKeys []string
+
+	flagChecks          string
+	flagExclude         string
+	flagMaxFetchedCerts int
+}
+
+type PKIHealthCheckHelper func(mount string) int
+
+func (c *PKIHealthCheckCommand) Synopsis() string {
+	return "Check PKI Secrets Engine health and operational status"
+}
+
+func (c *PKIHealthCheckCommand) Help() string {
+	helpText := `
+Usage: vault pki health-check [options] MOUNT
+
+  Reports status of the mount against best practices and pending failures,
+  including default TTL period of roles. This is an informative command and
+  not meant to be automatically parsed.
+
+  To check the pki-root mount:
+
+      $ vault pki health-check /pki-root
+
+  To exclude specific checks:
+
+      $ vault pki health-check -exclude=role-localhost-issuance /pki-root
+
+  Known checks:
+	  crl-validity            - checks if the CRL is still valid
+	  direct-issuance         - checks if a root is directly issuing leaf certs
+      issuer-validity         - checks if the issuing CA certificate is still valid
```

Comment:
You probably want to fix the indentation.

Reply:
Ah good catch, ty!